pkg/mgrconfig/config.go

```diff
@@ -38,8 +40,31 @@ type Config struct {
 	//	"qemu_args": "-fda {{TEMPLATE}}/fd"
 	WorkdirTemplate string `json:"workdir_template"`
 	// Directory with kernel object files (e.g. `vmlinux` for linux)
-	// (used for report symbolization and coverage reports, optional).
+	// (used for report symbolization, coverage reports and in tree modules finding, optional).
 	KernelObj string `json:"kernel_obj"`
+	// Directories for unstripped kernel module object files (optional)
+	// It's needed for out-of-tree module build in order to find ko files automatically
+	// If not all out-of-tree modules in this directory, use KernelModules for these
+	// exception cases
+	ModuleObj []string `json:"module_obj"`
+	// Map of kernel module object files and load address (optional).
+	// path is unstripped module obj path and
+	// addr (uint64) is the module load address on target,
+	//   ex 'wlan 16384 0 - Live 0xffffffffc0163000',
+	//   addr is 18446744072637263872 of 0xffffffffc0163000 above.
+	// disabled set to true to disable coverage for the module.
+	// For linux target, the addr is getting from /proc/modules
+	//   automatically during syz-fuzzer setup.
+	// Example config:
+	// "kernel_modules": [
+	//	{
+	//		"name": "wlan",
+	//		"path": "/var/lib/modules/unstripped/wlan.ko",
+	//		"addr": 18446744072637911040,
+	//		"disabled": true
+	//	}
+	// ]
+	KernelModules []backend.KernelModule `json:"kernel_modules"`
```

Comment:
I saw most of the functions like getReportGenerator is passing cfg instead of mgr, so I still this KernelModules in config, but I can remove all it's manual input. acceptable for you?

Reply:
The manager config is an important, user-visible thing with backward compatibility requirement. Users read it, try to understand, it produces cognitive load on users, later we can't simply change it because users may already use these parameters.
We absolutely should not put things into the config just because it makes it a little bit easier to pass things around somewhere deep in implementation details of the system.

So, yes, pass it separately.
If there are too many parameters, we can introduce a special object for them just to simplify passing them all around, like:
https://github.com/google/syzkaller/blob/master/pkg/build/build.go#L22-L35